pkg/identity/identity.go

```diff
@@ -96,5 +104,10 @@ func (id *Identity) GetModel() *models.Identity {
 
 // NewIdentity creates a new identity
 func NewIdentity(id NumericIdentity, lbls labels.Labels) *Identity {
-	return &Identity{ID: id, Labels: lbls}
+	var lblArray labels.LabelArray
+
+	if lbls != nil {
+		lblArray = lbls.ToSlice()
```

Comment:
I think `.ToSlice` is nil safe, but we probably use nil to denote some implicit state somewhere, don't we?

Reply:
Probably; I did not try to optimize because I wanted to port the exact behavior over from `Consumable`. Once everything is gathered into more logical units, I think we can then clearly define what certain types of state (nil, len(0), non-nil, etc.) denote in the documentation of the types.